src/mpid/ch4/netmod/ofi/ofi_am_events.h

```diff
@@ -21,7 +21,7 @@ MPL_STATIC_INLINE_PREFIX uint16_t MPIDI_OFI_am_get_next_recv_seqno(fi_addr_t add
     r = MPIDIU_map_lookup(MPIDI_OFI_global.am_recv_seq_tracker, id);
     if (r == MPIDIU_MAP_NOT_FOUND) {
         MPL_DBG_MSG_FMT(MPIDI_CH4_DBG_GENERAL, VERBOSE,
-                        (MPL_DBG_FDEST, "First time adding recv seqno addr=0x%016lx\n", addr));
+                        (MPL_DBG_FDEST, "First time adding recv seqno addr=%p\n", (void *) addr));
```

Comment:
I'm not sure these `fi_addr_t` values are better represented as `void *`. According to the libfabric developer guide 

> The fi_addr_t (fabric interface address datatype) is a 64-bit value that is used in all ‘fast-path’ operations – data transfers and completions.

and `fabric.h` public header

> typedef uint64_t                fi_addr_t;

I think `PRIu64` is probably better.

Reply:
I was trying to not rely on the fact that `fi_addr_t` is `uint64_t` because it's meant to be opaque.

Reply:
Yes, but on 32-bit platforms, the new code will also give a warning:

> autotest@bb52:/tmp % gcc -I $HOME/software/libfabric/sockets/include foo.c
> foo.c: In function 'main':
> foo.c:7:30: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
>    printf("fi_addr_t = %p\n", (void *)foobar);

```
#include <rdma/fabric.h>
#include <stdio.h>

int main(void) {
  fi_addr_t foobar;

  printf("fi_addr_t = %p\n", (void *)foobar);

  return 0;
}
```

Reply:
Yeah, that's true.  I guess `PRIu64` is the lesser of two evils.  I'll update the patch.